src/main/java/org/mskcc/oncokb/controller/MongoController.java

```diff
@@ -0,0 +1,341 @@
+            // create a new collection "trial_query" to save matched trials.
+            Boolean isDropped = MongoUtil.dropCollection(this.mongoDatabase, "trial_query");
+            if (matchedResults.size() > 0) {
+                if(isDropped) {
+                    Boolean isCreated = MongoUtil.createCollection(this.mongoDatabase,
+                        "trial_query", new ArrayList<>(matchedResults));
+                    if (!isCreated) {
+                        return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+                    }
+                }
+            }
+
+            String clinicalPath = FileUtil.buildJsonTempFile(clinicalArray.toString(), clinicalTempFile);
+            String genomicPath = FileUtil.buildJsonTempFile(genomicArray.toString(), genomicTempFile);
+
+            ProcessBuilder loadPb = new ProcessBuilder("python", System.getenv("CATALINA_HOME") +
+                "/webapps/matchminer-curate/WEB-INF/classes/matchminer-engine/matchengine.py",
+                "load", "-c", clinicalPath, "-g", genomicPath, "--patient-format", "json", "--query",
+                "--mongo-uri", this.uri);
+            Boolean isLoad = PythonUtil.runPythonScript(loadPb);
+
+            if(isLoad) {
+                // run MatchEngine match() with "--query" flag
+                ProcessBuilder matchPb = new ProcessBuilder("python", System.getenv("CATALINA_HOME") +
+                    "/webapps/matchminer-curate/WEB-INF/classes/matchminer-engine/matchengine.py", "match", "--query",
+                    "--mongo-uri", this.uri);
+                Boolean isMatch = PythonUtil.runPythonScript(matchPb);
+
+                if(isMatch) {
+                    // export matched result from collection "trial_match" in MongoDB "matchminer"
+                    List<Document> matchedTrialDocs = MongoUtil.getCollection(this.mongoDatabase,
+                        "new_trial_match");
+                    matchedResults.addAll(matchedTrialDocs);
+                } else {
+                    return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+            } else {
+                return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+            }
+
+            trialMatchResult = capsulateTrialMatchDoc(matchedResults);
+
+            clinicalTempFile.delete();
+            genomicTempFile.delete();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        return new ResponseEntity<>(trialMatchResult,  HttpStatus.OK);
+    }
+
+    public List<Genomic> annotateOncokbVariant(List<Genomic> genomics, String annotateApi) {
+
+        List<Genomic> results = new ArrayList<>(genomics);
+
+        try {
+            Set<MatchVariant> matchVariants = new HashSet<>();
+            List<Query> queries = new ArrayList<>();
+            Map<String, Set<String>> oncoVaMap = new HashMap<>();
```

Comment:
Key is hugo_symbol. Value is oncokb_variant but it might be an array. That's why I use set to remove duplicates of oncokb_variant and save it as an array in mongodb even if it only has 1 element.